### PR TITLE
Adding a new function named user_can_for_blog(), and unit tests for it.

### PR DESCRIPTION
**Pull request by Rajinsharwar**
Adding a new function named user_can_for_blog(), and unit tests for it.

Trac ticket: https://core.trac.wordpress.org/ticket/45197

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by johnbillion — CHANGES_REQUESTED** [review]
I've added some more tests for this change and the tests are failing. The function is incorrectly returning true when checking the `edit_posts` capability for a user who does not have access to the site that's being checked.

I'm not yet sure if this is a problem with the tests or the function.

**Comment by johnbillion** [comment]
@Rajinsharwar @JJJ Any thoughts on what's up here? The `test_user_can_for_blog()` test is failing when it checks the `edit_posts` capability of a user on a site that the user doesn't belong to. The result is incorrectly `true` instead of `false`.

File: tests/phpunit/tests/user/capabilities.php [hunk 1]
```diff
@@ -1671,12 +1676,66 @@ public function test_current_user_can_for_blog() {
...
+			return;
+		}
+
+		$blog_id = self::factory()->blog->create( array( 'user_id' => $user->ID ) );
+
+		$this->assertNotWPError( $blog_id );
+		$this->assertTrue( user_can_for_blog( $user->ID, $blog_id, 'edit_posts' ) );
+		$this->assertFalse( user_can_for_blog( $user->ID, $blog_id, 'foo_the_bar' ) );
+
+		$author = self::$users['author'];
```
Comment (Rajinsharwar) [inline tests/phpunit/tests/user/capabilities.php:1723]:
```suggestion
		$author = self::factory()->user->create( array( 'role' => 'author' ) );
```

**Comment by Rajinsharwar** [comment]
Fixed @johnbillion, used the user_can() instead in the function, the previous method didn't check for the current_blog I bielive.

**Review by johnbillion — APPROVED** [review]

**Comment by johnbillion** [comment]
https://core.trac.wordpress.org/changeset/59123

**Pull request closed without merging**